src/internal/hooks/useMessageReceivers.ts

```diff
@@ -31,7 +31,13 @@ export const useCommonMessageReceivers = (
 
   // Theme from Content
   const [themeData, setThemeData] = useState<ThemeData>();
-  const [themeDataFetched, setThemeDataFetched] = useState<boolean>(false); // needed because themeData can be empty
+  const [themeDataFetched, setThemeDataFetched] = useState<boolean>(
+    !templateMetadata?.themeEntityId
+  ); // needed because themeData can be empty
+
+  useEffect(() => {
+    setThemeDataFetched(!templateMetadata?.themeEntityId);
```

Comment:
Isn't this already handled on line 74?

Reply:
not exactly

if themeEntityId doesn't exist, then we want to setThemeDataFetched to true

the call to getThemeData will not occur if themeEntityId does not exist

There's also the situation where the page is loading
ThemeEntityId is 0
We setThemeDataFetched to true (because themeEntityId is 0)
ThemeEntityId updates to an actual id
We now want to have themeDataFetched set to false

This useEffect handles this case because whenever themeEntityId changes we update the themeDataFetched boolean

Reply:
I looked into why we don't need to do this for the veData and it's because we always make the query. I think we should do the same for theme to be consistent. So in ThemeEditorMessenger.tsx (alpha) we remove the themeEntityId check on `enabled: iFrameLoaded > 0 && !!themeEntityId,` and if themeEntityId is nullish then we just return nothing. That way on this side themeDataFetched will just get set without all of this extra logic.

Reply:
https://gerrit.yext.com/c/alpha/+/275129